### DISCUSSION TITLE
RabbitMQ Streams Support?

### DISCUSSION THREAD

**Discussion by GiorgioG** [rabbitmq/rabbitmq-dotnet-client#1058] [General]
Curious as to if/when we might see support for Streams feature in the .NET client.

https://blog.rabbitmq.com/posts/2021/07/rabbitmq-streams-overview/

**Comment by michaelklishin** [comment]
This client can consume from streams today but will never implement the stream protocol. 

You can help develop a [stream protocol client](https://github.com/rabbitmq/rabbitmq-stream-java-client) in .NET. We have more docs coming for client library developers closer towards RabbitMQ 3.9.0 final release.

**Reply by scott-g-roberts** [reply]
I'm confused by this reply. You say the .Net client will "never" implement the stream protocol but then ask for help developing it. Can you clarify what stream functionality will be available in the .Net client?

**Reply by scott-g-roberts** [reply]
Oh, perhaps you mean it will be a separate client SDK?

**Reply by GiorgioG** [reply]
Sounds more like it's going to be up to the community to develop a client.  Disappointing. If you need stream support and you're using .NET, you're better off using Kakfa, sadly.

**Reply by lhoguin** [reply]
This client is AMQP and AMQP can be used for streams but not with the same performance. There needs to be a separate client for the Stream protocol. We are looking into a .NET client (as well as clients for many other languages) but how soon it can be done will depend on prioritisation.

**Reply by kjnilsson** [reply]
We intend to develop a .NET client for the stream protocol. Follow this repository https://github.com/rabbitmq/rabbitmq-stream-dotnet-client for now and hopefully we will have more news / activity soon.

**Reply by 0UserName** [reply]
I understand correctly that the current client can read messages arriving in the stream, but re-reading the stream is impossible? Roughly speaking, only queue semantics are supported

**Comment by kjnilsson** [comment]
Hi,

No that isn't correct, please see:
https://www.rabbitmq.com/streams.html#consuming

Cheers
Karl

On Tue, 16 Nov 2021 at 15:55, 0UserName ***@***.***> wrote:

> I understand correctly that the current client can read messages arriving
> in the stream, but re-reading the stream is impossible? Roughly speaking,
> only queue semantics are supported
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-dotnet-client/discussions/1058#discussioncomment-1652356>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJAHFEGENZFWJLW7WJUXFLUMJ5GZANCNFSM5AKOWTCA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
*Karl Nilsson*

**Reply by 0UserName** [reply]
Indeed.. thanks for the tip.
